base/compiler/optimize.jl

```diff
@@ -33,6 +33,10 @@ const IR_FLAG_NOTHROW     = 0x01 << 5
 
 const TOP_TUPLE = GlobalRef(Core, :tuple)
 
+const MAX_INLINE_COST = typemax(UInt16)
+inlineable(src::Union{CodeInfo, Vector{UInt8}}) = ccall(:jl_ir_inlining_cost, UInt16, (Any,), src) != MAX_INLINE_COST
```

Comment:
How about parameterizing `inlineable` with `AbstractInterpreter`?
Then we can store whatever cost in `inline_cost` and each `AbstractInterpreter` can perform inlining of same code cache with their own threshold.

Reply:
This is a good idea, but yes, let me work this into a separate PR. Just because a couple other PRs will rely on this one and I'd like to keep the complexity low to start

Reply:
x-ref: https://github.com/JuliaGPU/GPUCompiler.jl/pull/335 where we want to customize inlining for GPU compilation